### DISCUSSION TITLE
Help: `Could not find the function with a single unnamed json or jsonb parameter in the schema cache`?

### DISCUSSION THREAD

**Discussion by jimmiejackson414** [supabase/supabase#7852] [Questions] [answered]
I'm trying to build an .rpc() function to update a jsonb column in my public.users table. I am building it through the Supabase UI, and it saves the function fine (no errors).

<img width="673" alt="Screen Shot 2022-07-22 at 11 39 25 AM" src="https://user-images.githubusercontent.com/93394140/180503019-f77fc7b4-f65c-4bd4-b894-77021f1d445a.png">

I've tried several different versions of this function. When I attempt to use this function from my graphql server, I receive an error:

```
{
  error: {
    message: 'Could not find the public.update_onboarding_steps(id, onboarding_steps) function or the public.update_onboarding_steps function with a single unnamed json or jsonb parameter in the schema cache',
    hint: 'If a new function was created in the database with this name and parameters, try reloading the schema cache.'
  }
}
```

and here is the way that I am using it:
```javascript
const { data, error } = await supabase
  .rpc('update_onboarding_steps', {
    onboarding_steps: JSON.stringify(rest.onboarding_steps),
    id
  })
  .single();
console.log({ data });
console.log({ error });
```

Is it an issue with the database function itself? It seems odd that it would save fine, and I can view the API documentation on how to use it, however I receive an error telling me that the function can't be found?

**Comment by steve-chavez** [comment, marked as answer]
> Could not find the public.update_onboarding_steps(id, onboarding_steps) function 

You defined your function with `id_input` and `val_input` parameters(not `id`/`onboarding_steps`), so it should be called as

```js
const { data, error } = await supabase
  .rpc('update_onboarding_steps', {
    id_input: 'val',
    val_input: 'val'
  });
```

**Reply by jimmiejackson414** [reply]
Wow haha. I need more coffee. Thanks!
